### PR TITLE
Support swapped DP/DM pins

### PR DESCRIPTION
**Pull request by sekigon-gonnoc**
implement #29

* Add new configuration option `pinout`
  * For example,
    *  pin_dp=1, pinout=PIO_USB_PINOUT_DPDM (default) > pin_dp=1, pin_dm = 2
    *  pin_dp=1, pinout=PIO_USB_PINOUT_DMDP (default) > pin_dp=1, pin_dm = 0
  * To set pinout, define `PIO_USB_PINOUT_DEFAULT` as `PIO_USB_PINOUT_DPDM`(default)/`PIO_USB_PINOUT_DMDP`, or set pio_usb_configuration.pinout`

### CODE REVIEW

File: src/pio_usb_configuration.h [hunk 1]
```diff
@@ -1,23 +1,33 @@
...
-    uint8_t sm_tx;
-    uint8_t tx_ch;
-    uint8_t pio_rx_num;
-    uint8_t sm_rx;
-    uint8_t sm_eop;
-    void* alarm_pool;
-    int8_t debug_pin_rx;
-    int8_t debug_pin_eop;
+  uint8_t pin_dp;
+  PIO_USB_PINOUT pinout;
```
Comment (hathach) [inline src/pio_usb_configuration.h:11]: maybe having `pin_dm` directly instead, if it is not set e.g `0`, we set it to `dp+1` as default. Just my opinion, it is up to you to decide which is best approach.

**Comment by machdyne** [comment]
FYI this is working for me. It would be best to be able to set DP and DM to any pin but requiring them to be next to each other isn't so bad as long as they can be swapped IMO. This is very helpful. If anyone needs hardware for testing PIO USB please let me know I may be able to provide some boards that might be useful. Thanks.

**Comment by ebadger** [comment]
I take it this change makes it possible to specify arbitrary GPIO pins for data +/-  as long as the pins are consecutive?
Will this pull request be completed?   I'm working on a project where I'd like to connect a USB type A port to 2 of the GPIO pins such that the micro USB can be used for power delivery without concern with data pins.  Does this change make my scenario possible?  Thanks

**Comment by ATMakersBill** [comment]
Just to give some more context, I made the mistake of assuming that A0 and A1 on the Adafruit QT Py RP2040 board would be consecutive and increasing.  They are actually GP28 and GP27 respectively.  It was a naïve mistake, and I feel foolish, but I am pretty sure that others will make it.  In the meantime, I have designed a board (and had 50 made) that is a perfect form factor but has the pins reversed.  This patch would solve that for me.  Thanks for doing the work and considering the pull request

**Comment by ATMakersBill** [comment]
image

**Comment by adirobu84** [comment]
What's the status of this PR? This is a very helpful feature. Like others have mentioned already, the need for DP and DM to be consecutive and DM = DP + 1 is not a constraint that is clear. In my case, I have a board where layout was optimized to swap DP and DM for signal integrity reasons.
Anything I can do to help this move along?

**Comment by eivindbohler** [comment]
Got this branch working with my custom board, but I see there's been lots of commits to main after this pr was made. Any chance of getting it merged? 😊

**Pull request closed without merging**